addons/account/static/tests/reconciliation_tests.js

```diff
@@ -1620,5 +1622,135 @@ QUnit.module('account', {
 
         clientAction.destroy();
     });
+
+    QUnit.only('Tax on account receivable', function(assert){
+        /*
+        Ecrire le test pour la reconciliation_model pour mon fix (loadTaxes)
+        faut faire le workflow complet, apparemment en js ça irait (lul) mais
+        je vais me péter le dents sur la logique fonctionnelle haha
+         */
+        assert.expect(17);
+
+        var reconcileNo;
+        var clientAction = new ReconciliationClientAction.ManualAction(null, this.params.options);
+        testUtils.addMockEnvironment(clientAction, {
+            data: this.params.data,
+            debug: true,  // console.log les rpc
+            session: {},
+            mockRPC: function(route, args) {
+                if (args.method === "name_search") {
+                    switch (args.model) {
+                        // mock the default mock to do the minimal processing required
+                        // to get the available values for the droplists.
+                        case 'account.account':
+                            assert.ok(true, "Accounts list has been fetched");
+                            return $.when(
+                                _.map(this.data[args.model].records, function (record) {
+                                    return [record.id, record.name];
+                                })
+                            );
+                        case 'account.tax':
+                            assert.ok(true, "Taxes list has been fetched");
+                            return $.when(
+                                _.map(this.data[args.model].records, function (record) {
+                                    return [record.id, record.display_name];
+                                })
+                            );
+                        case 'account.journal':
+                            assert.ok(true, "Journals list has been fetched");
+                            return $.when(
+                                _.map(this.data[args.model].records, function (record) {
+                                    return [record.id, record.display_name];
+                                })
+                            );
+                    }
+                }
+                if (args.method === 'process_move_lines') {
+                    var mv_line_ids = args.args[0][0].mv_line_ids.slice(0);
+                    console.log(mv_line_ids)
+                    mv_line_ids.sort(function(a, b) {return a - b});
+
+                    switch (reconcileNo) {
+                        case 1:
```

Comment:
We already discussed that I don't like it very much

since the first reconciliation is covered by the actual purpose of this test, It should be wise to not test the first step (by eliminating the the widget/the line or by directly selecting the second widget)

Tests have to be as simple as possible, with a clear purpose